File: Cafe-Server/MenuItem.cs
using System;
using System.Collections.Generic;

namespace Cafe_Server;

public partial class MenuItem
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string Weight { get; set; } = null!;

    public string Ingredients { get; set; } = null!;

    public string Caloric { get; set; } = null!;

    public double Price { get; set; }

    public bool? Available { get; set; }

    public int CategoryId { get; set; }

    public virtual Category Category { get; set; } = null!;
}

File: cafe_client/lib/models/menu_item.dart
class MenuItem {
  final int id;
  final String name;
  final int weight;
  final String ingredients;
  final int caloric;
  final double price;
  final bool available;
  final int categoryId;
  final String? imageLink;

  MenuItem(this.id, this.name, this.weight, this.ingredients, this.caloric,
      this.price, this.available, this.categoryId, this.imageLink);

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "name": name,
      "weight": weight.toString(),
      "ingredients": ingredients,
      "caloric": caloric.toString(),
      "price": price,
      "available": available,
      "categoryId": categoryId,
      "imageLink": imageLink,
    };
  }
}
